ivy/functional/ivy/meta.py

```diff
@@ -533,58 +533,95 @@ def fomaml_step(
 @handle_array_function
 def reptile_step(
     batch: ivy.Container,
-    cost_fn: Callable,
+    cost_fn: Callable[[ivy.Container, ivy.Container], ivy.Array],
     variables: ivy.Container,
     inner_grad_steps: int,
     inner_learning_rate: float,
     /,
     *,
-    inner_optimization_step: Callable = gradient_descent_update,
+    inner_optimization_step: Callable[
+        [Callable[[ivy.Array, ivy.Container], ivy.Array], ivy.Container, float],
+        ivy.Container,
+    ] = gradient_descent_update,
```

Comment:
I appreciate the detail you've added to these type hints, but I think we should leave them as just 'Callable' so they are less verbose and follow the same pattern as other functions.